--- a/src/wallet/main.rs	2024-09-06 12:05:03
+++ b/src/wallet/main.rs	2024-10-10 13:41:05
@@ -9,11 +9,14 @@
 use crate::shim::address::{CurrentNetwork, Network};
 use clap::Parser;
 use std::str::FromStr;
+use antithesis_sdk::prelude::*;
+use serde_json::json;
 
 pub fn main<ArgT>(args: impl IntoIterator<Item = ArgT>) -> anyhow::Result<()>
 where
     ArgT: Into<OsString> + Clone,
 {
+    antithesis_init();
     // Capture Cli inputs
     let Cli {
         opts,
