diff --git a/Makefile b/Makefile
index 901c8dc00..34682dda7 100644
--- a/Makefile
+++ b/Makefile
@@ -84,6 +84,7 @@ butterflynet: build-devnets
 interopnet: GOFLAGS+=-tags=interopnet
 interopnet: build-devnets
 
+lotus: GOFLAGS+=-race
 lotus: $(BUILD_DEPS)
 	rm -f lotus
 	$(GOCC) build $(GOFLAGS) -o lotus ./cmd/lotus
@@ -91,6 +92,7 @@ lotus: $(BUILD_DEPS)
 .PHONY: lotus
 BINS+=lotus
 
+lotus-miner: GOFLAGS+=-race
 lotus-miner: $(BUILD_DEPS)
 	rm -f lotus-miner
 	$(GOCC) build $(GOFLAGS) -o lotus-miner ./cmd/lotus-miner
